cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/transaction/stream/StreamAdminTest.java

```diff
@@ -287,6 +285,18 @@ public void testOwner() throws Exception {
       // expected
     }
 
+    // trying to create same stream with different owner should fail
+    properties.put(Constants.Security.PRINCIPAL, "someOtherUser/someHost@somekdc.net");
+    try {
+      streamAdmin.create(stream, properties);
+      Assert.fail("Should have failed to add the same stream with different owner");
+    } catch (IllegalArgumentException e) {
```

Comment:
I will fix this in this issue: https://issues.cask.co/browse/CDAP-8708 as it will touch this part.